sympy/series/fourier.py

```diff
@@ -440,16 +438,213 @@ def __sub__(self, other):
         return self.__add__(-other)
 
 
-class FiniteFourierSeries(Basic):
-    def __new__(cls, *args):
-        obj = Basic.__new__(cls, *args)
-        return obj
+class FiniteFourierSeries(FourierSeries):
+    r"""Represents Finite Fourier sine/cosine series.
 
-    def truncate(self, n=3):
-        return Add(*self._args)
+       For how to compute Fourier series, see the :func:`fourier_series`
+       docstring.
+
+       Parameters
+       ==========
+       f : Expr
+            Expression for finding fourier_series
+
+       limits : ( x, start, stop)
+            x is the independent variable for the expression f
+            (start, stop) is the period of the fourier series
+
+       exprs: (a0, an, bn) or Expr
+            a0 is the constant term a0 of the fourier series
+            an is a dictionary of coefficients of cos terms
+             `an[k] = coefficient of cos(pi*(k/L)*x)`
+            bn is a dictionary of coefficients of sin terms
+             `bn[k] = coefficient of sin(pi*(k/L)*x)`
+
+            or exprs can be an expression to be converted to fourier form
+
+       Methods
+       =======
+       This class is an extension of FourierSeries class.
+       Please refer to sympy.series.fourier.FourierSeries for
+       further information.
+
+       See Also
+       ========
+
+       sympy.series.fourier.FourierSeries
+       sympy.series.fourier.fourier_series
+       """
+
+    def __new__(cls, f, limits, exprs):
+
+        if not (type(exprs) == tuple and len(exprs) == 3):  # exprs is not of form (a0, an, bn)
+
+            # Converts the expression to fourier form
+            exprs = exprs.as_coeff_add()
+            exprs = exprs[0] + Add(*[TR10(i) for i in exprs[1]])
+            exp_ls = exprs.expand(trig=False, power_base=False, power_exp=False, log=False).as_coeff_add()
+
+            a0 = exp_ls[0]
+            x = limits[0]
+            L = abs(limits[2] - limits[1]) / 2
+
+            a = Wild('a', properties=[lambda k: k.is_Integer, lambda k: k != S.Zero, ])
+            b = Wild('b', properties=[lambda k: x not in k.free_symbols or k == S.Zero, ])
```

Comment:
Is `or k == S.Zero` necessary? `k = S.Zero` has no free symbols.

Reply:
I also think it is unnecessary.